src/master-playlist-controller.js

```diff
@@ -852,42 +853,36 @@ export class MasterPlaylistController extends videojs.EventTarget {
    * @return {TimeRange} the current time
    */
   setCurrentTime(currentTime) {
-    let buffered = Ranges.findRange(this.tech_.buffered(), currentTime);
+    const media = this.masterPlaylistLoader_ && this.masterPlaylistLoader_.media();
+    const segments = media && media.segments || [];
 
-    if (!(this.masterPlaylistLoader_ && this.masterPlaylistLoader_.media())) {
-      // return immediately if the metadata is not ready yet
-      return 0;
+    // Don't start segment loaders at the new location if:
+    // * master media is not ready
+    // * master media has no segments
+    if (!media || !segments.length) {
+      return currentTime;
     }
 
-    // it's clearly an edge-case but don't thrown an error if asked to
-    // seek within an empty playlist
-    if (!this.masterPlaylistLoader_.media().segments) {
-      return 0;
-    }
+    ['audio', 'subtitle', 'main'].forEach(function(type) {
+      // main segment loader is always active
+      // subtitle and audio segment loaders need to check
+      // for an activePlaylistLoader in mediaTypes_
+      if (type !== 'main' && !this.mediaTypes_[type.toUpperCase()].activePlaylistLoader) {
+        return;
+      }
 
-    // In flash playback, the segment loaders should be reset on every seek, even
-    // in buffer seeks. If the seek location is already buffered, continue buffering as
-    // usual
-    // TODO: redo this comment
-    if (buffered && buffered.length) {
-      return currentTime;
-    }
+      const loader = this[`${type}SegmentLoader_`];
 
-    // cancel outstanding requests so we begin buffering at the new
-    // location
-    this.mainSegmentLoader_.resetEverything();
-    this.mainSegmentLoader_.abort();
-    if (this.mediaTypes_.AUDIO.activePlaylistLoader) {
-      this.audioSegmentLoader_.resetEverything();
-      this.audioSegmentLoader_.abort();
-    }
-    if (this.mediaTypes_.SUBTITLES.activePlaylistLoader) {
-      this.subtitleSegmentLoader_.resetEverything();
-      this.subtitleSegmentLoader_.abort();
-    }
+      // if this loader does not have the requested time buffered
+      // reset it, abort current requests, and load at the new location.
+      if (!isBuffered(loader.buffered_(), currentTime)) {
```

Comment:
OK I think I figured out the only change we need now. Seems like the stutter always happens due to a `resetEverything` right after playback starts. This happens because the `buffered` value from tech doesn't tell you what the loader itself has buffered. It is a combination of all of the `buffered` values from our `SourceBuffers`. So when we try to seek to a `currentTime` of `0` if the `tech` buffered did not match all buffers were reset. Now we only reset buffers if the loaders `buffered` value does not contain currentTime.

Reply:
So, for each loader type, figure out if we have the time we're seeking to buffered, otherwise, reset that loader?

Reply:
yeah